### DISCUSSION TITLE
Generic precision configuration for Climate entities

### DISCUSSION THREAD

**Discussion by FlorianOosterhof** [home-assistant/architecture#1028] [Entity Models]
Currently each `Climate` entity has its own way to decide which `precision` and/or `target_temperature_step` values are used, and if there is choice for the user, how the user is allowed to set this.

My proposal is to allow the user to set these values for each climate entity through UI in a generic way, i.e. not via the UI configuration of the integration that creates the climate entity, e.g. such as is now the case in the `opentherm_gw` integration, but instead generically in a "climate entity configuration window", similarly to how each sensor (at least those which have a `unique_id`) are allowed to set the display precision.

A consideration I had is that sometimes an integration knows more than the user, e.g. a many thermostats only allow to set the target temperature to a multiple of 0.5. Then a `target_temperature_step` of 0.1 does not make sense, and in my opinion the user should be protected from this by not allowing 0.1 to be set on the climate entity of that integration; allowing it would make the user think it will now use and/or see values with that precision, whereas the device will still not actually be using/exposing that precision. I would imagine this might save a bunch of issues/tickets from users.

Finally, I propose to extend the set of allowed `PRECISION_*` values to include at leasts `HUNDRETHS`, but perhaps even `THOUSANDS`. The reason is (quite selfishly) that for instance the `opentherm_gw` integration reads data from the OpenTherm protocol, in which most temperatures are f8.8 values, i.e. multiples of 1/256, in which case the `HUNDRETHS` option would allow the current temperature to be much more accurately represented in the climate entity without having to resort to the sensor that contains the underlying value. `THOUSANDS` could potentially be added because then it will unlock the full precision of the OpenTherm protocol, though I could understand that this would be considered overkill for a climate entity.

Personal note: in my opinion the `HUNDRETHS` option should anyway be added, because for any feedback-controlled system to work properly, a good resolution feedback-signal must be present, so it can be assumed that most thermostats (not only those that support OpenTherm) actually know the room temperature value to such a precision. I know mine does, and I am interested in that kind of precision.

**Comment by gjohansson-ST** [comment]
I guess precision makes sense and could maybe align with `sensor`.
However I don't see what's good with making it allowed for the user to change the `target_temperature_step` more than the complexity it's going to bring.
